arbnode/resourcemanager/resource_management.go

```diff
@@ -95,25 +104,22 @@ func isSupported(c limitChecker) bool {
 	return err == nil
 }
 
-// newLimitChecker attempts to auto-discover the mechanism by which it
-// can check system limits. Currently Cgroups V1 and V2 are supported.
-// If no supported mechanism is discovered, it logs an error and
-// fails open, ie it creates a trivialLimitChecker that does no checks.
-func newLimitChecker(conf *Config) limitChecker {
+// newCgroupsMemoryLimitCheckerIfSupported attempts to auto-discover whether
+// Cgroups V1 or V2 is supported for checking system memory limits.
+func newCgroupsMemoryLimitCheckerIfSupported(conf *Config) (*cgroupsMemoryLimitChecker, error) {
```

Comment:
nit: go tends to encourage shorter names. You can drop "IfSupported" suffix and just add in the comment above the function that this returns errNotSupported if it's not supported.

Reply:
I'm not sure what to rename it to since `newCgroupsMemoryLimitChecker` is already the factory method for the `CgroupsMemoryLimitChecker` and this version handles auto-detecting cgroups v1 or v2.